File: app/src/main/java/io/github/gianpamx/bicitoday/di/AppComponent.java
package io.github.gianpamx.bicitoday.di;

import javax.inject.Singleton;

import dagger.Component;
import io.github.gianpamx.bicitoday.recording.di.RecordingComponent;
import io.github.gianpamx.bicitoday.recording.di.RecordingModule;

@Component(modules = {AppModule.class})
@Singleton
public interface AppComponent {
    RecordingComponent recordingComponent(RecordingModule recordingModule);
}

File: app/src/main/java/io/github/gianpamx/bicitoday/di/AppComponent.kt
package io.github.gianpamx.bicitoday.di

import dagger.Component
import io.github.gianpamx.bicitoday.record.di.RecordComponent
import io.github.gianpamx.bicitoday.record.di.RecordModule
import javax.inject.Singleton

@Singleton
@Component(modules = arrayOf(AppModule::class))
interface AppComponent {
    fun recordComponent(recordModule: RecordModule) : RecordComponent
}
